### PR TITLE
Pandas1.3 32bit

### LINKED ISSUE #8169: 32bit test suite: pandas 1.3.3 "Buffer dtype mismatch"
**What happened**:

When running the test suite in a 32-bit environment on the openSUSE buildservice, I get the following error

```
[  672s] =================================== FAILURES ===================================
[  672s] ______________________ test_categorical_set_index[tasks] _______________________
[  672s] [gw1] linux -- Python 3.9.7 /usr/bin/python3.9
[  672s] 
[  672s] shuffle = 'tasks'
[  672s] 
[  672s]     @pytest.mark.parametrize("shuffle", ["disk", "tasks"])
[  672s]     def test_categorical_set_index(shuffle):
[  672s]         df = pd.DataFrame({"x": [1, 2, 3, 4], "y": ["a", "b", "b", "c"]})
[  672s]         df["y"] = pd.Categorical(df["y"], categories=["a", "b", "c"], ordered=True)
[  672s]         a = dd.from_pandas(df, npartitions=2)
[  672s]     
[  672s]         with dask.config.set(scheduler="sync", shuffle=shuffle):
[  672s]             b = a.set_index("y", npartitions=a.npartitions)
[  672s]             d1, d2 = b.get_partition(0), b.get_partition(1)
[  672s] >           assert list(d1.index.compute()) == ["a"]
[  672s] 
[  672s] /usr/lib/python3.9/site-packages/dask/dataframe/tests/test_categorical.py:274: 
[  672s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[  672s] /usr/lib/python3.9/site-packages/dask/base.py:288: in compute
[  672s]     (result,) = compute(self, traverse=False, **kwargs)
[  672s] /usr/lib/python3.9/site-packages/dask/base.py:570: in compute
[  672s]     results = schedule(dsk, keys, **kwargs)
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:563: in get_sync
[  672s]     return get_async(
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:506: in get_async
[  672s]     for key, res_info, failed in queue_get(queue).result():
[  672s] /usr/lib/python3.9/concurrent/futures/_base.py:438: in result
[  672s]     return self.__get_result()
[  672s] /usr/lib/python3.9/concurrent/futures/_base.py:390: in __get_result
[  672s]     raise self._exception
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:548: in submit
[  672s]     fut.set_result(fn(*args, **kwargs))
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:237: in batch_execute_tasks
[  672s]     return [execute_task(*a) for a in it]
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:237: in <listcomp>
[  672s]     return [execute_task(*a) for a in it]
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:228: in execute_task
[  672s]     result = pack_exception(e, dumps)
[  672s] /usr/lib/python3.9/site-packages/dask/local.py:223: in execute_task
[  672s]     result = _execute_task(task, data)
[  672s] /usr/lib/python3.9/site-packages/dask/core.py:121: in _execute_task
[  672s]     return func(*(_execute_task(a, cache) for a in args))
[  672s] /usr/lib/python3.9/site-packages/dask/dataframe/shuffle.py:861: in shuffle_group
[  672s]     return group_split_dispatch(df, c, k, ignore_index=ignore_index)
[  672s] /usr/lib/python3.9/site-packages/dask/utils.py:576: in __call__
[  672s]     return meth(arg, *args, **kwargs)
[  672s] /usr/lib/python3.9/site-packages/dask/dataframe/backends.py:358: in group_split_pandas
[  672s]     indexer, locations = pd._libs.algos.groupsort_indexer(
[  672s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[  672s] 
[  672s] >   ???
[  672s] E   ValueError: Buffer dtype mismatch, expected 'const intp_t' but got 'long long'
[  672s] 
[  672s] pandas/_libs/algos.pyx:194: ValueError
```

If I didn't overlook something, it is the same error for 187 tests:
```
[  681s] =========================== short test summary info ============================
[  681s] FAILED dataframe/tests/test_categorical.py::test_categorical_set_index[tasks]
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-False-1-1] - ...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-False-1-4] - ...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-False-3-1] - ...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-False-3-4] - ...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-True-1-1] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-True-1-4] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-True-3-1] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[False-True-3-4] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-False-1-1] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-False-1-4] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-False-3-1] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-False-3-4] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_value_counts_with_normalize - ...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-True-1-1] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_value_counts_with_normalize_and_dropna
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-True-1-4] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-True-3-1] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_series_map[True-True-3-4] - Va...
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_reduction_split[split_out]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_apply_tasks - ValueError...
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column0-2]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column0-3]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column1-2]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column1-3]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column2-2]
[  681s] FAILED dataframe/tests/test_groupby.py::test_groupby_split_out_multiindex[column2-3]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-2-1] - ...
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-2-4] - ...
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-2-20]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-5-1] - ...
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-5-4] - ...
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[5-5-20]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-2-1]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-2-4]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-2-20]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-5-1]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-5-4]
[  681s] FAILED dataframe/tests/test_groupby.py::test_hash_groupby_aggregate[20-5-20]
[  681s] FAILED dataframe/tests/test_groupby.py::test_split_out_multi_column_groupby
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[idx-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[idx-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[idx-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[idx-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on1-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on1-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on1-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on1-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on2-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on2-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on2-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on2-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on3-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on3-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on3-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_unknown[on3-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[idx-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[idx-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[idx-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[idx-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on1-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on1-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on1-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on2-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on1-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on2-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on2-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on2-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on2-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on3-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on3-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on2-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on3-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on2-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on3-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on3-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_right[idx-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on3-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on3-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_right[on1-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_known[on3-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[idx-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[idx-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_right[on2-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[idx-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_right[on3-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[idx-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[idx-True-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on1-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on1-left]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[idx-0.75-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on1-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on1-True-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on1-outer]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on1-0.75-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_unknown_to_unknown[on2-inner]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on2-True-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on2-0.75-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on3-True-right]
[  681s] FAILED dataframe/tests/test_merge_column_and_index.py::test_merge_known_to_double_bcast_left[on3-0.75-right]
[  681s] FAILED dataframe/tests/test_multi.py::test_hash_join[tasks-inner] - ValueErro...
[  681s] FAILED dataframe/tests/test_multi.py::test_hash_join[tasks-left] - ValueError...
[  681s] FAILED dataframe/tests/test_multi.py::test_hash_join[tasks-right] - ValueErro...
[  681s] FAILED dataframe/tests/test_multi.py::test_hash_join[tasks-outer] - ValueErro...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge[tasks-inner] - ValueError: B...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge[tasks-outer] - ValueError: B...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge[tasks-left] - ValueError: Bu...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge[tasks-right] - ValueError: B...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_index_patterns[inner-tasks]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_index_patterns[outer-tasks]
[  681s] FAILED tests/test_distributed.py::test_fused_blockwise_dataframe_merge[True]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_index_patterns[left-tasks]
[  681s] FAILED tests/test_distributed.py::test_fused_blockwise_dataframe_merge[False]
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-2-1] - Val...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-2-4] - Val...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-2-20] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-5-1] - Val...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-5-4] - Val...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[5-5-20] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-2-1] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-2-4] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-2-20] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-5-1] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-5-4] - Va...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_hash_split_unique[20-5-20] - V...
[  681s] FAILED dataframe/tests/test_dataframe.py::test_split_out_drop_duplicates[None]
[  681s] FAILED dataframe/tests/test_dataframe.py::test_split_out_drop_duplicates[2]
[  681s] FAILED dataframe/tests/test_dataframe.py::test_split_out_value_counts[None]
[  681s] FAILED dataframe/tests/test_dataframe.py::test_split_out_value_counts[2] - Va...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_index_patterns[right-tasks]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_multiple_columns[tasks-inner]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_multiple_columns[tasks-outer]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_multiple_columns[tasks-left]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_by_multiple_columns[tasks-right]
[  681s] FAILED tests/test_distributed.py::test_combo_of_layer_types - ValueError: Buf...
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_index_without_divisions[tasks]
[  681s] FAILED dataframe/tests/test_multi.py::test_half_indexed_dataframe_avoids_shuffle
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[lg-28-left]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[lg-28-right]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[lg-32-left]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[lg-32-right]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[sm-28-left]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[sm-28-right]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[sm-32-left]
[  681s] FAILED dataframe/tests/test_multi.py::test_merge_tasks_large_to_small[sm-32-right]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle[tasks] - ValueError: Buf...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle_npartitions_task - Value...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle_npartitions_lt_input_partitions_task
[  681s] FAILED dataframe/tests/test_shuffle.py::test_index_with_non_series[tasks] - V...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_index_with_dataframe[tasks] - Va...
[  681s] FAILED tests/test_distributed.py::test_shuffle_priority - ValueError: Buffer ...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle_from_one_partition_to_one_other[tasks]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle_empty_partitions[tasks]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_set_index_tasks[4] - ValueError:...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_set_index_tasks[7] - ValueError:...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_set_index_tasks[23] - ValueError...
[  681s] FAILED tests/test_distributed.py::test_map_partitions_df_input - ValueError: ...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_set_index_tasks_2[tasks] - Value...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_set_index_tasks_3[tasks] - Value...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_shuffle_sort[tasks] - ValueError...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_rearrange[threads-tasks] - Value...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_rearrange[processes-tasks] - Val...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_gh_2730 - ValueError: Buffer dty...
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-id-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-id-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-id-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-name-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-name-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-name-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on2-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on2-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on2-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on3-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on3-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[None-on3-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-id-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-id-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-id-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-name-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-name-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-name-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on2-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on2-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on2-False]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on3-None]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on3-True]
[  681s] FAILED dataframe/tests/test_shuffle.py::test_dataframe_shuffle_on_tasks_api[4-on3-False]
...
[  682s] = 187 failed, 7973 passed, 891 skipped, 119 xfailed, 8 xpassed, 4462 warnings, 5 rerun in 636.78s (0:10:36) =
```

Full buildlog: 
[dask-test-i586_log.txt](https://github.com/dask/dask/files/7210987/dask-test-i586_log.txt)

```python
pytest --pyargs dask -rfEs -n auto
```

**Anything else we need to know?**:
- Pandas 1.3.3
- Distributed is built without --cythonize for 32-bit (https://github.com/dask/dask/issues/7489#issuecomment-812474850)

https://github.com/dask/dask/blob/7b661440e7a916b8a9b369cc24bd4180f8564e0b/dask/dataframe/backends.py#L356-L360

**Environment**:

- Dask version: 2019.9.1
- Python version: 3.9.7
- Operating System: openSUSE Tumbleweed
- Install method (conda, pip, source): source (rpm package)

### PR DESCRIPTION
**Pull request by bnavigator**
As requested by @mgorny, https://github.com/dask/dask/issues/8169#issuecomment-1074393142.

- [x] Closes #8169
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

I haven't tested with an older pandas, but I think this raises the minimum version to 1.3

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: setup.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 extras_require: dict[str, list[str]] = {
     "array": ["numpy >= 1.18"],
     "bag": [],  # keeping for backwards compatibility
-    "dataframe": ["numpy >= 1.18", "pandas >= 1.0"],
+    "dataframe": ["numpy >= 1.18", "pandas >= 1.3"],
```
Comment (jcrist) [inline setup.py:17]: I'm not sure if we want to bump the minimum pandas version here. 1.3 came out ~9 months ago. Is there a way to make these changes backwards compatible?

File: continuous_integration/environment-mindeps-dataframe.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - toolz=0.8.2
   # optional dependencies pulled in by pip install dask[dataframe]
   - numpy=1.18
-  - pandas=1.0
+  - pandas=1.3
```
Comment (jcrist) [inline continuous_integration/environment-mindeps-dataframe.yaml:16]: Same here

**Comment by jsignell** [comment]
ok to test

**Pull request merged**